extensions/wikia/AdEngine/js/video/player/porvata/floaterConfiguration.js

```diff
@@ -45,18 +56,39 @@ define('ext.wikia.adEngine.video.player.porvata.floaterConfiguration', [
 					var elements = floatingContext.elements;
 
 					elements.viewport = elements.adContainer;
+					floatingContext.forceDoNotFloat();
 				},
 				calculateViewportHeight: function (viewport) {
 					return win.getComputedStyle(viewport).height;
 				},
 				floatLater: function (callback) {
+					var floatingContext = this,
+						video = this.elements.video;
+
 					if (this.isOutsideOfViewport()) {
-						callback(this);
-						this.fireEvent('start');
+						floatingContext.fireEvent('start');
 					}
+
+					video.addEventListener('loaded', function () {
+						floatingContext.floatAgain();
+						if (floatingContext.isOutsideOfViewport()) {
+							callback(floatingContext);
+						}
+					});
 				},
 				onAttach: function (floatingContext) {
-					floatingContext.elements.viewport.style.removeProperty('height');
+					var elements = floatingContext.elements;
+
+					elements.viewport.style.removeProperty('height');
+					elements.iframe.style.removeProperty('height');
+				},
+				onBeforeDetach: function (floatingContext) {
+					var viewport = floatingContext.elements.viewport;
+
+					viewport.style.height = win.getComputedStyle(viewport).height;
+				},
+				onDetach: function (floatingContext) {
+					floatingContext.elements.iframe.style.height = floatingContext.getHeight() + 'px';
```

Comment:
discussed offline - indeed, is visible that the values are modified in pairs. We can re-thing this approach when introducing next integration